### DISCUSSION TITLE
Unable to update Nominatim from version 4.2.2 to 4.3.2

### DISCUSSION THREAD

**Discussion by xfasterx** [osm-search/Nominatim#3300] [Q&A]
**Describe the bug**

Downloaded and unpacked Nominatim-4.3.2.tar.bz2

Did try to update from version 4.2.2 to version 4.3.2 on local server with command:
 
nominatim admin --migrate 

**Software Environment (please complete the following information):**
- Nominatim version: 4.2.2
- OS:  Ubuntu 22.04

nominatim admin --migrate

Using project directory: /var/lib/postgresql
Checking for necessary database migrations
Database already at latest version (4.2.2-0)

Database already at latest version (4.2.2-0) notes that upgrade is not needed bacause it states its the latest version. Since it is not how do i update to 4.3.2 ?

**Comment by lonvia** [comment]
Please describe the exact steps you took and the command output you got.

**Comment by xfasterx** [comment]
Periodicly i update the Nominatim database with the command: nominatim replication.

And untill now that always worked correctly. But last time i did the process exited with this error (see image).

psycopg2.errors.InternalError_: line_substring: 1st arg isn't a line CONTEXT: PL/pgSQL function osmline_update() line 113 at assignment

![Update error](https://cdn.imgpaste.net/2023/11/16/KUowAY.jpg)

When i restart the update process it will start the process again and apperently stop again at the same point.

Since i did not find a solution to fix that i decided to try to upgrade the Nominatim server.

Im on 4.2.2
So i downloaded the latest update and put it in the same install directory /srv/nominatim where i originally installed 4.2.2 from.

In this directory: /srv/nominatim
Nominatim-4.2.2.tar.bz2
Nominatim-4.3.2.tar.bz2

Did unpack: tar xf Nominatim-4.3.2.tar.bz2

Did Run the command: nominatim admin --migrate 
From
/srv/nominatim
/srv/nominatim/build
/srv/nominatim/Nominatim-4.2.2
/srv/nominatim/Nominatim-4.3.2

Output
: Using project directory: /var/lib/postgresql
: Checking for necessary database migrations
: Database already at latest version (4.2.2-0)

**Comment by mtmail** [comment]
Downloading and unpacking the *.tar.bz2 isn't enough. You need to build and install Nominatim (https://nominatim.org/release-docs/latest/admin/Installation/#building-nominatim). Then `nominatim --version` should print the new version. With the new version you can run `nominatim admin --migrate`
